doc/format_specifications.rdoc

```diff
@@ -163,34 +176,123 @@ Format +argument+ as a binary integer:
 
   sprintf('%b', 1)  # => "1"
   sprintf('%b', 4)  # => "100"
+
   # Prefix '..' for negative value.
   sprintf('%b', -4) # => "..100"
 
-Type-specific modifier:
+  # Alternate format.
+  sprintf('%#b', 4)  # => "0b100"
+  sprintf('%#B', 4)  # => "0B100"
 
-- '#' flag' (use alternate format):
+==== Specifier +c+
 
-    sprintf('%#b', 1)  # => "0b1"
-    sprintf('%#B', 1)  # => "0B1"
-    sprintf('%#b', 4)  # => "0b100"
+Format +argument+ as a single character:
 
-==== Specifier +c+
+  sprintf('%c', 'A') # => "A"
+  sprintf('%c', 65)  # => "A"
 
 ==== Specifier +d+
 
+Format +argument+ as a decimal integer:
+
+  sprintf('%d', 100)  # => "100"
+  sprintf('%d', -100) # => "-100"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +e+ and +E+
 
+Format +argument+ in
```

Comment:
Do we want to mention that for `e`/`E`, `f`, `g`/`G` the precision can be specified and defaults to 6.

For example:

```ruby
irb(main):109:0> sprintf('%e', 1.0000001)
=> "1.000000e+00"
irb(main):110:0> sprintf('%.7e', 1.0000001)
=> "1.0000001e+00"
irb(main):111:0> sprintf('%f', 1.0000001)
=> "1.000000"
irb(main):112:0> sprintf('%.7f', 1.0000001)
=> "1.0000001"
```